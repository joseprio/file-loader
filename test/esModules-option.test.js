import webpack from './helpers/compiler';

describe('when applied with `esModules` option', () => {
  it('matches snapshot without value', async () => {
    const config = {
      loader: {
        test: /(png|jpg|svg)/,
      },
    };

    const stats = await webpack('fixture.js', config);
    const [module] = stats.toJson().modules;
    const { assets, source } = module;

    expect({ assets, source }).toMatchSnapshot();
  });

  it('matches snapshot for `true` value', async () => {
    const config = {
      loader: {
        test: /(png|jpg|svg)/,
        options: {
          esModules: true,
        },
      },
    };

    const stats = await webpack('fixture.js', config);
    const [module] = stats.toJson().modules;
    const { assets, source } = module;

    expect({ assets, source }).toMatchSnapshot();
  });

  it('matches snapshot for `false` value', async () => {
    const config = {
      loader: {
        test: /(png|jpg|svg)/,
        options: {
          esModules: false,
        },
      },
    };

    const stats = await webpack('fixture.js', config);
    const [module] = stats.toJson().modules;
    const { assets, source } = module;

    expect({ assets, source }).toMatchSnapshot();
  });
});
