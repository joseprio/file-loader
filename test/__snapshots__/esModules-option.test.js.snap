// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when applied with \`esModules\` option matches snapshot for \`false\` value 1`] = `
Object {
  "assets": Array [
    "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = __webpack_public_path__ + \\"9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;

exports[`when applied with \`esModules\` option matches snapshot for \`true\` value 1`] = `
Object {
  "assets": Array [
    "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "export default __webpack_public_path__ + \\"9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;

exports[`when applied with \`esModules\` option matches snapshot without value 1`] = `
Object {
  "assets": Array [
    "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = __webpack_public_path__ + \\"9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;
